src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -82,6 +82,10 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
         if spec.spdk_mem_size:
             daemon_spec.extra_files['spdk_mem_size'] = str(spec.spdk_mem_size)
 
+        # Indicate to the daemon whether to enable grpc tracing
+        if spec.grpc_trace:
+            daemon_spec.extra_files['grpc_trace'] = str(spec.grpc_trace)
```

Comment:
I think doing this is as part of  `extra_files` is a bit odd given it's just a string we use to setup an env variable. We should probably just set `daemon_spec.final_config['grpc_trace']` instead. You can see how this is done for `crush_location` in the MonService class in mgr/cephadm/services/cephadmservice.py. You might want to add a `generate_config` function  for the NvmeofService class to handle this as well.